arrows/vxl/high_pass_filter.cxx

```diff
@@ -0,0 +1,490 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "high_pass_filter.h"
+
+#include <arrows/vxl/image_container.h>
+#include <vital/util/enum_converter.h>
+
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+#include <vil/vil_transpose.h>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+enum filter_mode
+{
+  MODE_box,
+  MODE_bidir,
+};
+
+ENUM_CONVERTER( mode_converter, filter_mode, { "box", MODE_box }, { "bidir",
+                                                                    MODE_bidir } )
+
+// --------------------------------------------------------------------------------------
+/// Private implementation class
+class high_pass_filter::priv
+{
+public:
+
+  priv( high_pass_filter* parent )
+    : p( parent )
+  {
+  }
+
+  ~priv()
+  {
+  }
+
+  // Internal parameters/settings
+  filter_mode mode = MODE_box;
+  size_t kernel_width = 7;
+  size_t kernel_height = 7;
+  bool treat_as_interlaced = false;
+  bool output_net_only = false;
+  high_pass_filter* p;
+
+  // Perform box filtering
+  template < typename PixType >
+  vil_image_view< PixType >
+  box_high_pass_filter( const vil_image_view< PixType >& grey_img )
+  {
+    vil_image_view< PixType > output;
+    output.set_size( grey_img.ni(), grey_img.nj(), 3 );
```

Comment:
Not strictly necessary, but I think these can be rewritten as:
```suggestion
    vil_image_view< PixType > output{ grey_img.ni(), grey_img.nj(), 3 };
```
...? (Unless this way is preferred for some reason?)